File: DevPlaygroundFrontEnd/src/types/todo.ts
export type TodoStatus = 'PENDING' | 'IN_PROGRESS' | 'COMPLETED' | 'CANCELLED'

export interface Todo {
  id: number
  userName: string
  title: string
  description: string
  status: TodoStatus
  dueDate: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateTodoRequest {
  userName: string
  title: string
  description: string
  status: TodoStatus | string
  dueDate: string | null
}

export const TodoStatusLabels: Record<TodoStatus, string> = {
  PENDING: '未着手',
  IN_PROGRESS: '進行中',
  COMPLETED: '完了',
  CANCELLED: 'キャンセル'
}

File: DevPlaygroundBackEnd/src/main/java/com/example/DevPlayground/entity/Todo.java
package com.example.DevPlayground.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "todos")
@Getter
@Setter
@NoArgsConstructor
public class Todo {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String userName;

    @Column(nullable = false, length = 30)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String status;

    private LocalDateTime dueDate;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    public Todo(String userName, String title, String description, String status, LocalDateTime dueDate) {
        this.userName = userName;
        this.title = title;
        this.description = description;
        this.status = status;
        this.dueDate = dueDate;
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}